src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,38 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null) {
+            return false;
+        }
+        if (object instanceof Rectangle) {
+            Rectangle rectangle = (Rectangle) object;
+
+            if ((width == rectangle.getWidth() || (width != null && rectangle.getWidth() != null
+                    && width.equals(rectangle.getWidth())))
+                    && (length == rectangle.getLength()
+                    || (length != null && rectangle.getLength() != null
+                    && length.equals(rectangle.getLength())))
+                    && (color == rectangle.getColor()
+                    || (color != null && rectangle.getLength() != null
+                    && color.equals(rectangle.getColor())))) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 9;
```

Comment:
why 9?

Reply:
only prime numbers...it is a mistake
done